Introduction to built-in data structures in Java

srishilesh opened the pull request:
### Article Checklist
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.

**NOTE: (1st time contributors - submit ONLY one article to get familiar with the process and style-once published, you can then submit any number of articles.)**
      
#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

This fixes #1174 .

cyberShaw commented on articles/introduction-to-built-in-data-structures-in-java/index.md:
```diff
@@ -0,0 +1,304 @@
+
+#### LinkedList
+
+Linked lists are linear data structures that are connected using pointers, and holds data. The memory allocation is not contiguous which makes it better than the use of Arrays.
```
> Linked lists are linear data structures that are connected using pointers, and holds data. 

Please include a one-liner which defines a pointer

cyberShaw commented on articles/introduction-to-built-in-data-structures-in-java/index.md:
```diff
@@ -0,0 +1,304 @@
+
+##### Remove Elements
+
+Similarly, for deletion of an element in a user-defined linked list is cumbersome, where the pointer of the previous element is pointed to the address of the next element. When working with huge data, working with pointers is not easy.
```
Could you give a quick example of deletion of an element in a linkedList using the standard method (without using in-built data structures)

cyberShaw commented on articles/introduction-to-built-in-data-structures-in-java/index.md:
```diff
@@ -0,0 +1,304 @@
+
+##### Adding Elements
+
+`HashMap` is one of the important data structures, where mapping the key and the value helps in accessing the elements with very little time complexity. To map a key with a value, we make use of the `put(key, value)` method.
```
> accessing the elements with very little time complexity.

It would be great if you can specify the exact average case time complexity of accessing an element in a HashMap.

cyberShaw commented on articles/introduction-to-built-in-data-structures-in-java/index.md:
````diff
@@ -0,0 +1,304 @@
+
+Example:
+
+```java
````
Include the example of popping an empty stack.

cyberShaw commented on articles/introduction-to-built-in-data-structures-in-java/index.md:
```diff
@@ -0,0 +1,304 @@
+
+#### Built-in vs. User-defined Data Structures
+
+Data structures like Stack, or Queue can be created on our own using other basic data structures. For example, Stacks can be created using Arrays or Linked lists. These are known as User-defined data structures.
```
```suggestion
Data structures like Stack or Queue can be created on our own using other basic data structures. For example, Stacks can be created using Arrays or Linked lists. These are known as User-defined data structures.
```

cyberShaw commented on articles/introduction-to-built-in-data-structures-in-java/index.md:
```diff
@@ -0,0 +1,304 @@
+
+##### Removing Elements
+
+In `Queue`, removing elements is referred to as "Polling". Polling can be performed on one end of the queue, called the "Rear" end. Similarly, adding can be performed on the opposite end, called the "Front" end.
```
```suggestion
In `Queue`, removing elements is referred to as "Polling". Polling can be performed on one end of the queue, called the "Rear" end. Similarly, adding can be performed on the opposite end called the "Front" end.
```

cyberShaw reviewed (changes requested):
Great article, @srishilesh! I've suggested some changes to make the article more polished. Please make these changes and we can move on to final review.

srishilesh commented on articles/introduction-to-built-in-data-structures-in-java/index.md:
````diff
@@ -0,0 +1,304 @@
+
+```java
+list1.set(1, "Welcome"); // Replaces "World" with "Welcome" in list1
+// OUTPUT: list1 = ["Hello", "Welcome"]
````
Hi @cyberShaw . Thanks for the suggestion.

As you mentioned, adding output statements using `System.out.println()` is not possible for certain data structures, the data structure must be iterated using a for loop or any other iterator. In that case, should I write a function for iterating, and call it everytime for displaying the output?

cyberShaw replied:
Yeah, it would be great if you can do that. Beginners could easily just copy and paste the code to see the results and easily learn how the data structure works

cyberShaw commented:
I checked out the changes you made, @srishilesh. They look perfect. If you can make the changes that you haven't made yet (Adding the output statements, corner cases and the time complexity of HashMap) we can move on to the final review

srishilesh commented:
> I checked out the changes you made, @srishilesh. They look perfect. If you can make the changes that you haven't made yet (Adding the output statements, corner cases and the time complexity of HashMap) we can move on to the final review

Sure @cyberShaw. I was waiting for your opinion before proceeding. I'll complete them as you mentioned.

hectorkambow commented:
plagiarism check done

srishilesh commented:
Hi @cyberShaw ,

Sorry for the delay, I have completed the remaining suggestions. Please let me know if I have to make any more changes.

Changes made are:
- Fixed the grammatical errors are you mentioned
- Adding output for all the data structures (Except Stack & Queue - since printing them would make no sense)
- Added a few corner cases for all data structures - like removing elements from empty list
- Other minor fixes for the flow

cyberShaw commented:
> Hi @cyberShaw ,
> 
> Sorry for the delay, I have completed the remaining suggestions. Please let me know if I have to make any more changes.
> 
> Changes made are:
> 
> * Fixed the grammatical errors are you mentioned
> * Adding output for all the data structures (Except Stack & Queue - since printing them would make no sense)
> * Added a few corner cases for all data structures - like removing elements from empty list
> * Other minor fixes for the flow

Great job, @srishilesh!
You've made the necessary changes. We can move on to final review of this article.

hectorkambow commented:
@srishilesh  
Great article  - I think the community will find it really useful. 

Please be sure to go over the small but important changes I made to the flow of the content. 

Especially important since we are going to be reviewing articles as well right. 

cc @cyberShaw  Please do the same ⏫  
Thank you 🚀
